File: src/main/java/ru/trukhmanov/twochairsbackend/repository/UserRepository.java
package ru.trukhmanov.twochairsbackend.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import ru.trukhmanov.twochairsbackend.entity.User;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long>{
    Optional<User> findByPhoneNumber(String phoneNumber);
}

File: src/main/kotlin/ru/trukhmanov/twochairsbackend/repository/UserRepository.kt
package ru.trukhmanov.twochairsbackend.repository

import org.springframework.data.jpa.repository.JpaRepository
import ru.trukhmanov.twochairsbackend.entity.User
import java.util.Optional

interface UserRepository : JpaRepository<User, Long> {
    fun findByPhoneNumber(phoneNumber: String): Optional<User>
}
